### PR TITLE
Add H.263 support.

### PR DESCRIPTION
**Pull request by dreamaniajp**
Hi, 
Since current mp4parse-rust doesn't handle s263 and d263 boxes, I tried to add some codes to handle it.
Could you help to review this?

Thanks a lot!

### CODE REVIEW

**Comment by SingingTree** [comment]
Hi, sorry for the delay in looking at this. I'm hoping to go through this week, but it's been a little hectic. If I don't comment by next week please feel free to ping me to remind me.

**Comment by dreamaniajp** [comment]
> Hi, sorry for the delay in looking at this. I'm hoping to go through this week, but it's been a little hectic. If I don't comment by next week please feel free to ping me to remind me.

Sure, thanks a lot for reviewing!

File: mp4parse/tests/public.rs [hunk 1]
```diff
@@ -32,6 +32,7 @@ static IMAGE_AVIF_CORRUPT: &str = "tests/bug-1655846.avif";
 static IMAGE_AVIF_CORRUPT_2: &str = "tests/bug-1661347.avif";
 static IMAGE_AVIF_GRID: &str = "av1-avif/testFiles/Microsoft/Summer_in_Tomsk_720p_5x4_grid.avif";
 static AVIF_TEST_DIR: &str = "av1-avif/testFiles";
+static VIDEO_H263_3GP: &str = "tests/bbb_sunflower_QCIF_30fps_h263_noaudio_1f.3gp";
```
Comment (SingingTree) [inline mp4parse/tests/public.rs:35]: Could you add a brief comment describing how the file was created to help with reproducing similar files if needed?

**Review by SingingTree — COMMENTED** [review]
Looks reasonable to me. Like various other codecs we're just storing the data as a blob rather than parsing it into something structured -- so this behaves similar to existing code for stuff like AVC where the code using the parser will need to impose structure on the data if it wants.

File: mp4parse/tests/public.rs (hunk 1 above)
Comment (dreamaniajp) [reply mp4parse/tests/public.rs:37]: Sure, just add the comment in the updated commit, thanks.

**Comment by dreamaniajp** [comment]
Hi @SingingTree , thanks for your review.  
I updated the comment about how to produce the testing file.  
Thanks!

**Comment by SingingTree** [comment]
Hey @dreamaniajp, could you check to make sure you've pushed the changes? I'm not seeing them on the PR.

**Comment by dreamaniajp** [comment]
Hi @SingingTree 
The change is in
https://github.com/mozilla/mp4parse-rust/pull/245/files#diff-c38a967fc4de8c38b6426bcd04e0db7b8dea5bcd157e6bcbc4cab6c117f980c7R40

Thanks!.

**Comment by SingingTree** [comment]
My apologies. I just noticed that a comment I thought I'd submitted was still pending, I've submitted it now.

File: mp4parse/tests/public.rs (hunk 1 above)
Comment (dreamaniajp) [reply mp4parse/tests/public.rs:46]: Indeed, the command is more generic and way better, thanks!
I've updated this information and new test file in the latest commit.
Thanks again!

**Review by SingingTree — APPROVED** [review]

**Comment by SingingTree** [comment]
Thanks for the PR!

**Comment by baumanj** [comment]
FYI, this will require a change on the mozilla-central side of things:
```
 0:33.08 /Users/jbauman/src/mozilla-unified/dom/media/mp4/MP4Metadata.cpp:293:13: error: enumeration value 'MP4PARSE_CODEC_H263' not handled in switch [-Werror,-Wswitch]
 0:33.08     switch (codecType) {
```

**Pull request merged**